// @stripe/ui-extension-sdk/ui/@stripe-internal/tailor-dashboard-components/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabPanel = exports.Tab = exports.SignInView = exports.StripeFileUploader = exports.SettingsView = exports.Link = exports.Img = exports.FocusView = exports.ContextView = exports.Sparkline = exports.LineChart = exports.BarChart = void 0;
const react_1 = require("@remote-ui/react");
exports.BarChart = (0, react_1.createRemoteReactComponent)("BarChart");
exports.LineChart = (0, react_1.createRemoteReactComponent)("LineChart");
exports.Sparkline = (0, react_1.createRemoteReactComponent)("Sparkline");
exports.ContextView = (0, react_1.createRemoteReactComponent)("ContextView", {
    fragmentProps: [
        "actions",
        "banner",
        "primaryAction",
        "secondaryAction",
        "footerContent",
    ],
});
exports.FocusView = (0, react_1.createRemoteReactComponent)("FocusView", {
    fragmentProps: ["primaryAction", "secondaryAction", "footerContent"],
});
exports.Img = (0, react_1.createRemoteReactComponent)("Img");
exports.Link = (0, react_1.createRemoteReactComponent)("Link");
exports.SettingsView = (0, react_1.createRemoteReactComponent)("SettingsView");
exports.StripeFileUploader = (0, react_1.createRemoteReactComponent)("StripeFileUploader");
exports.SignInView = (0, react_1.createRemoteReactComponent)("SignInView", {
    fragmentProps: ["descriptionActionContents", "footerContent"],
});
exports.Tab = (0, react_1.createRemoteReactComponent)("Tab");
exports.TabPanel = (0, react_1.createRemoteReactComponent)("TabPanel");
