// @stripe/ui-extension-sdk/testing/matchers/props.js
"use strict";
/*
Copyright 2020-present, Shopify Inc.
Copyright 2022-present, Stripe Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.toHaveProps = void 0;
const jest_matcher_utils_1 = require("jest-matcher-utils");
const utilities_1 = require("./utilities");
function toHaveProps(node, props) {
    (0, utilities_1.assertIsNode)(node, {
        expectation: 'toHaveProps',
        isNot: this.isNot,
    });
    if (props == null || typeof props !== 'object') {
        return {
            pass: false,
            message: () => `You passed ${props == null ? String(props) : `a ${typeof props}`} as props, but it must be an object.`,
        };
    }
    const pass = Object.keys(props).every((key) => this.equals(props[key], node.props[key]));
    const message = pass
        ? () => `${(0, jest_matcher_utils_1.matcherHint)('.not.toHaveProps', node.toString())}\n\n` +
            `Expected the React element:\n  ${(0, jest_matcher_utils_1.RECEIVED_COLOR)(node.toString())}\n` +
            `Not to have props:\n  ${(0, jest_matcher_utils_1.printExpected)(props)}\n` +
            `Received:\n  ${(0, jest_matcher_utils_1.printReceived)(node.props)}\n`
        : () => {
            const diffString = (0, utilities_1.diffPropsForNode)(node, props, {
                expand: this.expand,
            });
            return (`${(0, jest_matcher_utils_1.matcherHint)('.toHaveProps', node.toString())}\n\n` +
                `Expected the React element:\n  ${(0, jest_matcher_utils_1.RECEIVED_COLOR)(node.toString())}\n` +
                `To have props:\n  ${(0, jest_matcher_utils_1.printExpected)(props)}\n` +
                `Received:\n  ${(0, jest_matcher_utils_1.printReceived)(node.props)}\n${diffString ? `Difference:\n${diffString}\n` : ''}`);
        };
    return { pass, message };
}
exports.toHaveProps = toHaveProps;
