// @stripe/ui-extension-sdk/ui/component-props-test/LineChart.test-d.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
/* eslint-disable react/react-in-jsx-scope */
const index_1 = require("../index");
const lineChart = 'LineChart';
exports.default = {
    jsx: ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)(index_1.LineChart, { data: [
                    {
                        x: 'Mon',
                        y: 5,
                        type: 'product',
                    },
                ], x: "x", y: "y", color: "type", axis: "x", grid: "x", legend: false, tooltip: false }), (0, jsx_runtime_1.jsx)(index_1.LineChart, { data: [
                    {
                        x: 'Mon',
                        y: 5,
                        type: 'product',
                    },
                ], x: {
                    value: 'test',
                    range: [],
                    domain: [],
                    label: 'test',
                    format: 'capitalize',
                }, y: {
                    value: 'test',
                    range: [],
                    domain: [],
                    label: 'test',
                    format: { currency: 'USD' },
                }, color: {
                    value: 'test',
                    domain: [],
                    label: 'test',
                    format: { unit: 'acre-per-centimeter' },
                } })] })),
    remoteComponentTypes: { lineChart },
};
