// @stripe/ui-extension-sdk/ui/@sail/ui/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TailorDateField = exports.List = exports.ListItem = exports.Tooltip = exports.TextField = exports.TextArea = exports.TabPanels = exports.TabList = exports.Tabs = exports.TabPanel = exports.Tab = exports.Switch = exports.Spinner = exports.Select = exports.Radio = exports.Notice = exports.Menu = exports.MenuGroup = exports.MenuItem = exports.Link = exports.Icon = exports.FormFieldGroup = exports.Divider = exports.ChipList = exports.Chip = exports.Checkbox = exports.ButtonGroup = exports.Button = exports.BasicTableHeaderCell = exports.BasicTableCell = exports.BasicTableFooter = exports.BasicTableRow = exports.BasicTableBody = exports.BasicTableHeader = exports.BasicTable = exports.Banner = exports.Badge = exports.AccordionItem = exports.Accordion = void 0;
const react_1 = require("@remote-ui/react");
exports.Accordion = (0, react_1.createRemoteReactComponent)("Accordion");
exports.AccordionItem = (0, react_1.createRemoteReactComponent)("AccordionItem", {
    fragmentProps: ["title", "subtitle", "media", "actions"],
});
exports.Badge = (0, react_1.createRemoteReactComponent)("Badge");
exports.Banner = (0, react_1.createRemoteReactComponent)("Banner", {
    fragmentProps: ["description", "title", "actions"],
});
exports.BasicTable = (0, react_1.createRemoteReactComponent)("BasicTable");
exports.BasicTableHeader = (0, react_1.createRemoteReactComponent)("BasicTableHeader");
exports.BasicTableBody = (0, react_1.createRemoteReactComponent)("BasicTableBody");
exports.BasicTableRow = (0, react_1.createRemoteReactComponent)("BasicTableRow");
exports.BasicTableFooter = (0, react_1.createRemoteReactComponent)("BasicTableFooter");
exports.BasicTableCell = (0, react_1.createRemoteReactComponent)("BasicTableCell");
exports.BasicTableHeaderCell = (0, react_1.createRemoteReactComponent)("BasicTableHeaderCell");
exports.Button = (0, react_1.createRemoteReactComponent)("Button");
exports.ButtonGroup = (0, react_1.createRemoteReactComponent)("ButtonGroup", {
    fragmentProps: ["menuTrigger"],
});
exports.Checkbox = (0, react_1.createRemoteReactComponent)("Checkbox", {
    fragmentProps: ["label"],
});
exports.Chip = (0, react_1.createRemoteReactComponent)("Chip");
exports.ChipList = (0, react_1.createRemoteReactComponent)("ChipList");
exports.Divider = (0, react_1.createRemoteReactComponent)("Divider");
exports.FormFieldGroup = (0, react_1.createRemoteReactComponent)("FormFieldGroup");
exports.Icon = (0, react_1.createRemoteReactComponent)("Icon");
exports.Link = (0, react_1.createRemoteReactComponent)("Link");
exports.MenuItem = (0, react_1.createRemoteReactComponent)("MenuItem");
exports.MenuGroup = (0, react_1.createRemoteReactComponent)("MenuGroup", {
    fragmentProps: ["title"],
});
exports.Menu = (0, react_1.createRemoteReactComponent)("Menu", {
    fragmentProps: ["trigger"],
});
exports.Notice = (0, react_1.createRemoteReactComponent)("Notice", {
    fragmentProps: ["description", "title", "actions"],
});
exports.Radio = (0, react_1.createRemoteReactComponent)("Radio", {
    fragmentProps: ["label"],
});
exports.Select = (0, react_1.createRemoteReactComponent)("Select", {
    fragmentProps: ["label"],
});
exports.Spinner = (0, react_1.createRemoteReactComponent)("Spinner");
exports.Switch = (0, react_1.createRemoteReactComponent)("Switch", {
    fragmentProps: ["label"],
});
exports.Tab = (0, react_1.createRemoteReactComponent)("Tab");
exports.TabPanel = (0, react_1.createRemoteReactComponent)("TabPanel");
exports.Tabs = (0, react_1.createRemoteReactComponent)("Tabs");
exports.TabList = (0, react_1.createRemoteReactComponent)("TabList");
exports.TabPanels = (0, react_1.createRemoteReactComponent)("TabPanels");
exports.TextArea = (0, react_1.createRemoteReactComponent)("TextArea", {
    fragmentProps: ["label"],
});
exports.TextField = (0, react_1.createRemoteReactComponent)("TextField", {
    fragmentProps: ["label"],
});
exports.Tooltip = (0, react_1.createRemoteReactComponent)("Tooltip", {
    fragmentProps: ["trigger"],
});
exports.ListItem = (0, react_1.createRemoteReactComponent)("ListItem", {
    fragmentProps: ["value", "title", "secondaryTitle", "icon", "image"],
});
exports.List = (0, react_1.createRemoteReactComponent)("List");
exports.TailorDateField = (0, react_1.createRemoteReactComponent)("TailorDateField", {
    fragmentProps: ["label"],
});
