// @stripe/ui-extension-sdk/ui/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableRow = exports.TableHeaderCell = exports.TableHead = exports.TableFooter = exports.TableCell = exports.TableBody = exports.Table = exports.TextField = exports.TextArea = exports.Switch = exports.Select = exports.Radio = exports.Link = exports.Inline = exports.Icon = exports.DateField = exports.Checkbox = exports.Button = exports.Box = exports.Tooltip = exports.Tabs = exports.TabPanels = exports.TabList = exports.Spinner = exports.MenuItem = exports.MenuGroup = exports.Menu = exports.ListItem = exports.List = exports.FormFieldGroup = exports.Divider = exports.ChipList = exports.Chip = exports.ButtonGroup = exports.Banner = exports.Badge = exports.AccordionItem = exports.Accordion = void 0;
// we shouldn't export * from @sail/ui because we want to rename some things,
// and we have manual versions of others.
var ui_1 = require("./@sail/ui");
Object.defineProperty(exports, "Accordion", { enumerable: true, get: function () { return ui_1.Accordion; } });
Object.defineProperty(exports, "AccordionItem", { enumerable: true, get: function () { return ui_1.AccordionItem; } });
Object.defineProperty(exports, "Badge", { enumerable: true, get: function () { return ui_1.Badge; } });
Object.defineProperty(exports, "Banner", { enumerable: true, get: function () { return ui_1.Banner; } });
// Button,
Object.defineProperty(exports, "ButtonGroup", { enumerable: true, get: function () { return ui_1.ButtonGroup; } });
// Checkbox,
Object.defineProperty(exports, "Chip", { enumerable: true, get: function () { return ui_1.Chip; } });
Object.defineProperty(exports, "ChipList", { enumerable: true, get: function () { return ui_1.ChipList; } });
Object.defineProperty(exports, "Divider", { enumerable: true, get: function () { return ui_1.Divider; } });
Object.defineProperty(exports, "FormFieldGroup", { enumerable: true, get: function () { return ui_1.FormFieldGroup; } });
// Icon,
// Link,
Object.defineProperty(exports, "List", { enumerable: true, get: function () { return ui_1.List; } });
Object.defineProperty(exports, "ListItem", { enumerable: true, get: function () { return ui_1.ListItem; } });
Object.defineProperty(exports, "Menu", { enumerable: true, get: function () { return ui_1.Menu; } });
Object.defineProperty(exports, "MenuGroup", { enumerable: true, get: function () { return ui_1.MenuGroup; } });
Object.defineProperty(exports, "MenuItem", { enumerable: true, get: function () { return ui_1.MenuItem; } });
// Notice is deprecated and should be removed as of major version 7 of the SDK
// Notice,
// Radio,
// Select,
Object.defineProperty(exports, "Spinner", { enumerable: true, get: function () { return ui_1.Spinner; } });
// Switch,
// Tab,
Object.defineProperty(exports, "TabList", { enumerable: true, get: function () { return ui_1.TabList; } });
// TabPanel,
Object.defineProperty(exports, "TabPanels", { enumerable: true, get: function () { return ui_1.TabPanels; } });
// BasicTable as Table,
// BasicTableBody as TableBody,
// BasicTableCell as TableCell,
// BasicTableFooter as TableFooter,
// BasicTableHeader as TableHead,
// BasicTableHeaderCell as TableHeaderCell,
// BasicTableRow as TableRow,
Object.defineProperty(exports, "Tabs", { enumerable: true, get: function () { return ui_1.Tabs; } });
// TailorDateField,
// TextArea,
// TextField,
Object.defineProperty(exports, "Tooltip", { enumerable: true, get: function () { return ui_1.Tooltip; } });
__exportStar(require("./@stripe-internal/tailor-dashboard-components"), exports);
var manual_components_1 = require("./manual_components");
Object.defineProperty(exports, "Box", { enumerable: true, get: function () { return manual_components_1.Box; } });
Object.defineProperty(exports, "Button", { enumerable: true, get: function () { return manual_components_1.Button; } });
Object.defineProperty(exports, "Checkbox", { enumerable: true, get: function () { return manual_components_1.Checkbox; } });
Object.defineProperty(exports, "DateField", { enumerable: true, get: function () { return manual_components_1.DateField; } });
Object.defineProperty(exports, "Icon", { enumerable: true, get: function () { return manual_components_1.Icon; } });
Object.defineProperty(exports, "Inline", { enumerable: true, get: function () { return manual_components_1.Inline; } });
Object.defineProperty(exports, "Link", { enumerable: true, get: function () { return manual_components_1.Link; } });
Object.defineProperty(exports, "Radio", { enumerable: true, get: function () { return manual_components_1.Radio; } });
Object.defineProperty(exports, "Select", { enumerable: true, get: function () { return manual_components_1.Select; } });
Object.defineProperty(exports, "Switch", { enumerable: true, get: function () { return manual_components_1.Switch; } });
Object.defineProperty(exports, "TextArea", { enumerable: true, get: function () { return manual_components_1.TextArea; } });
Object.defineProperty(exports, "TextField", { enumerable: true, get: function () { return manual_components_1.TextField; } });
Object.defineProperty(exports, "Table", { enumerable: true, get: function () { return manual_components_1.Table; } });
Object.defineProperty(exports, "TableBody", { enumerable: true, get: function () { return manual_components_1.TableBody; } });
Object.defineProperty(exports, "TableCell", { enumerable: true, get: function () { return manual_components_1.TableCell; } });
Object.defineProperty(exports, "TableFooter", { enumerable: true, get: function () { return manual_components_1.TableFooter; } });
Object.defineProperty(exports, "TableHead", { enumerable: true, get: function () { return manual_components_1.TableHead; } });
Object.defineProperty(exports, "TableHeaderCell", { enumerable: true, get: function () { return manual_components_1.TableHeaderCell; } });
Object.defineProperty(exports, "TableRow", { enumerable: true, get: function () { return manual_components_1.TableRow; } });
