// @stripe/ui-extension-sdk/ui/manual_components.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableHeaderCell = exports.TableCell = exports.TableFooter = exports.TableHead = exports.TableRow = exports.TableBody = exports.Table = exports.DateField = exports.Icon = exports.Button = exports.Link = exports.Checkbox = exports.Switch = exports.Radio = exports.Select = exports.TextArea = exports.TextField = exports.Inline = exports.Box = void 0;
const react_1 = require("@remote-ui/react");
exports.Box = (0, react_1.createRemoteReactComponent)('Box');
exports.Inline = (0, react_1.createRemoteReactComponent)('Inline');
exports.TextField = (0, react_1.createRemoteReactComponent)('TextField', {
    fragmentProps: ['label'],
});
exports.TextArea = (0, react_1.createRemoteReactComponent)('TextArea', {
    fragmentProps: ['label'],
});
exports.Select = (0, react_1.createRemoteReactComponent)('Select', {
    fragmentProps: ['label'],
});
exports.Radio = (0, react_1.createRemoteReactComponent)('Radio', {
    fragmentProps: ['label'],
});
exports.Switch = (0, react_1.createRemoteReactComponent)('Switch', {
    fragmentProps: ['label'],
});
exports.Checkbox = (0, react_1.createRemoteReactComponent)('Checkbox', {
    fragmentProps: ['label'],
});
exports.Link = (0, react_1.createRemoteReactComponent)('Link');
exports.Button = (0, react_1.createRemoteReactComponent)('Button');
exports.Icon = (0, react_1.createRemoteReactComponent)('Icon');
exports.DateField = (0, react_1.createRemoteReactComponent)('DateField', {
    fragmentProps: ['label'],
});
exports.Table = (0, react_1.createRemoteReactComponent)('Table');
exports.TableBody = (0, react_1.createRemoteReactComponent)('TableBody');
exports.TableRow = (0, react_1.createRemoteReactComponent)('TableRow');
exports.TableHead = (0, react_1.createRemoteReactComponent)('TableHead');
exports.TableFooter = (0, react_1.createRemoteReactComponent)('TableFooter');
exports.TableCell = (0, react_1.createRemoteReactComponent)('TableCell');
exports.TableHeaderCell = (0, react_1.createRemoteReactComponent)('TableHeaderCell');
